src/main/java/com/amazon/opendistroforelasticsearch/security/DefaultObjectMapper.java

```diff
@@ -69,9 +71,25 @@ public static void inject(final InjectableValues.Std injectableValues) {
         defaulOmittingObjectMapper.setInjectableValues(injectableValues);
     }
 
-    public static boolean getOrDefault(Map<String, Object> properties, String key, boolean defaultValue) {
-        Boolean value = (Boolean)properties.get(key);
-        return value != null ? value.booleanValue() : defaultValue;
+    public static boolean getOrDefault(Map<String, Object> properties, String key, boolean defaultValue) throws JsonProcessingException {
+        Object value = properties.get(key);
+        if (value == null) {
+            return defaultValue;
+        } else if (value instanceof Boolean) {
+            return (boolean)value;
+        } else if (value instanceof String) {
+            String text = ((String)value).trim();
```

Comment:
```suggestion
            String text = ((String)value).trim().toLower();
            if ("true".equals(text)) {
```

Reply:
The only lower case "true" and capitalized "True" are supported ("TRUE" and "tRue" are not valid) the same way how Jackson default deserializer works.